### PR TITLE
refactor: Convert development stage List and ListItem to SDS from MUI (#2179)

### PR DESCRIPTION
**Pull request by frano-m**
- #2179 

### Reviewers
**Functional:** 

@tihuan 

---

## Changes
- modified development stage filter MUI components to SDS

### CODE REVIEW

File: frontend/src/components/common/Filter/components/FilterMultiPanel/components/FilterPanelList/style.ts [hunk 1]
```diff
@@ -1,29 +1,45 @@
...
-    marginLeft: 22,
-  },
-});
-/* eslint-enable sort-keys -- ignore object key order for style objects */
+export const PanelSublist = styled(List)`
+  margin-left: 22px;
+`;
+
+export const PanelListItem = styled(ListItem)`
+  && {
```
Comment (frano-m) [inline frontend/src/components/common/Filter/components/FilterMultiPanel/components/FilterPanelList/style.ts:11]: Hey @tihuan what are your thoughts on this? I had to add the `&&` to obtain specificity over the SDS List Item styles. I'm wondering if there is a better way to do this?

Comment (tihuan) [reply frontend/src/components/common/Filter/components/FilterMultiPanel/components/FilterPanelList/style.ts:11]: Ah thanks so much for flagging this, @frano-m ! That's my bad 😆 I don't recall why I added extra specificity in `ListItem` in SDS, so I just created a PR to correct it!

https://github.com/chanzuckerberg/sci-components/pull/201

Comment (tihuan) [inline frontend/src/components/common/Filter/components/FilterMultiPanel/components/FilterPanelList/style.ts:11]:
```suggestion
  /* Remove && after upgrading SDS version to have this commit https://github.com/chanzuckerberg/sci-components/pull/201 */
  && {
```

**Review by tihuan — APPROVED** [review]
Thanks so much for using SDS `List` and `ListItem`, @frano-m 🤩 🙏 ! I created a PR in SDS to fix `ListItem`. Sorry for the bug!

File: frontend/src/components/common/Filter/components/FilterMultiPanel/components/FilterPanelList/style.ts (hunk 1 above)
Comment (frano-m) [reply frontend/src/components/common/Filter/components/FilterMultiPanel/components/FilterPanelList/style.ts:11]: Oh @tihuan I've been wondering why I was having a wrestling match ... thank you sooo much 🙏🏽 ... Happy days!

Comment (frano-m) [reply frontend/src/components/common/Filter/components/FilterMultiPanel/components/FilterPanelList/style.ts:11]: Legend @tihuan thank you!

**Pull request merged**